### PR TITLE
Unchecked responseBody could cause NullPointerException.

### PR DESCRIPTION
**Pull request by hoangpham95**
## Motivation
Unchecked `null` for the responseBody will cause NullPointerException.
When running my application, I observed this exception:
```
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String okhttp3.ResponseBody.string()' on a null object reference
       at com.facebook.react.modules.network.NetworkingModule$3.onResponse(Unknown Source)
       at okhttp3.RealCall$AsyncCall.execute(Unknown Source)
       at okhttp3.internal.NamedRunnable.run(Unknown Source)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
       at java.lang.Thread.run(Thread.java:818)
```

### CODE REVIEW

File: ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java [hunk 1]
```diff
@@ -361,7 +361,7 @@ public void onResponse(Call call, Response response) throws IOException {
 
               // Otherwise send the data in one big chunk, in the format that JS requested.
               String responseString = "";
-              if (responseType.equals("text")) {
```
Comment (janicduplessis) [inline ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java]: There are other cases below where we don't handle null responseBody could you make sure those are covered too?

**Comment by janicduplessis** [comment]
Do you know why responseBody can be null? When it is the case what is the proper thing to do? Should we show an error / warning or just return an empty string?

File: ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java (hunk 1 above)
Comment (hoangpham95) [inline ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java]: Sure. I'll add those changes.

**Comment by hoangpham95** [comment]
When running with a slow network or on a slow responsive server, my app throw a `NullPointerException`. I would doubt that the `Response response` object has empty content.

**Comment by janicduplessis** [comment]
@AaaChiuuu Could you have a look at this? I'm not sure what would be the proper way to handle this case.

**Comment by hoangpham95** [comment]
@janicduplessis any update on this PR?

File: ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java [hunk 2]
```diff
@@ -353,15 +353,15 @@ public void onResponse(Call call, Response response) throws IOException {
...
                 ResponseUtil.onRequestSuccess(eventEmitter, requestId);
                 return;
               }
 
               // Otherwise send the data in one big chunk, in the format that JS requested.
               String responseString = "";
-              if (responseType.equals("text")) {
+              if (responseType != null && responseType.equals("text") && responseBody != null) {
                 responseString = responseBody.string();
               } else if (responseType.equals("base64")) {
```
Comment (shergin) [inline ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java]: We have to check `responseType` nullability here as well. Or, even better, we can check nulabity somewhere above.

**Comment by shergin** [comment]
@janicduplessis Should we land this? Any thoughts about why `responseType` can be null?

**Comment by janicduplessis** [comment]
@shergin I'd like to understand in what case it is null to know how we should handle it. Should we treat the response body as text?

**Pull request closed without merging**